apps/engineeringblog/utils/index.ts

```diff
@@ -0,0 +1,56 @@
+import fs from "fs";
+import path from "path";
+import matter from "gray-matter";
+import { format } from "date-fns";
+export type Article = {
+  slug: string;
+  title: string;
+  description: string;
+  date: string;
+  featuredImage: string;
+  content: string;
+};
+
+export type ArticleWithoutContent = Omit<Article, "content">;
+
+export function getAllPosts(): ArticleWithoutContent[] {
```

Comment:
@DavidTheProgrammer ContentLayer is deprecated and [does not support NextjS 14](https://github.com/gsoft-inc/wl-hopper/issues/164) anymore. Rather than use another package, I decided to use the manual approach until we have a reason not to.

Reply:
Oh, is that so? Unfortunate. The manual approach is okay because we at least have gray-matter to parse the front-matter, which is key. 🚀 LGTM